pkg/drift/drift.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package drift
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/abcxyz/guardian/pkg/drift/assets"
+	"github.com/abcxyz/guardian/pkg/drift/iam"
+	"github.com/abcxyz/guardian/pkg/drift/terraform"
+)
+
+func Process(ctx context.Context, organizationID int64, bucketQuery string) error {
+	assetsClient, err := assets.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("error initializing assets Client: %w", err)
+	}
+	folders, err := assetsClient.GetHierarchyAssets(ctx, organizationID, "cloudresourcemanager.googleapis.com/Folder")
+	if err != nil {
+		return fmt.Errorf("unable to get folders: %w", err)
+	}
+	projects, err := assetsClient.GetHierarchyAssets(ctx, organizationID, "cloudresourcemanager.googleapis.com/Project")
+	if err != nil {
+		return fmt.Errorf("unable to get folders: %w", err)
+	}
+	buckets, err := assetsClient.GetBuckets(ctx, organizationID, bucketQuery)
+	if err != nil {
+		return fmt.Errorf("error fetching terraform state GCS buckets: %w", err)
+	}
+	fmt.Printf("Fetching IAM for %d Folders and %d Projects\n", len(folders), len(projects))
+
+	gcpIAM, err := getActualGCPIAM(ctx, organizationID, folders, projects)
+	if err != nil {
+		return fmt.Errorf("error determining GCP IAM: %w", err)
+	}
+	fmt.Printf("Fetching terraform state from %d Buckets\n", len(buckets))
+	tfIAM, err := getTerraformStateIAM(ctx, organizationID, folders, projects, buckets)
+	if err != nil {
+		return fmt.Errorf("error determining Terraform State: %w", err)
+	}
+	fmt.Printf("Found %d gcp IAM entries\n", len(gcpIAM))
+	fmt.Printf("Found %d terraform IAM entries\n", len(tfIAM))
+
+	clickOpsChanges := difference(gcpIAM, tfIAM)
+	missingTerraformChanges := difference(tfIAM, gcpIAM)
+
+	fmt.Printf("Found %d Click Ops Changes\n", len(clickOpsChanges))
+	fmt.Printf("Found %d Missing Terraform Changes\n", len(missingTerraformChanges))
+
+	return nil
+}
+
+func getActualGCPIAM(
+	ctx context.Context,
+	organizationID int64,
+	folders []assets.HierarchyNode,
+	projects []assets.HierarchyNode,
+) (map[string]iam.AssetIAM, error) {
```

Comment:
```suggestion
) (map[string]*iam.AssetIAM, error) {
```

unless that's an interface type